rt_scheduled_v_ran/scripts/rt_v_scheduled_routes.py

```diff
@@ -0,0 +1,108 @@
+import dask
+import dask.dataframe as dd
+import geopandas as gpd
+import pandas as pd
+from calitp_data_analysis.geography_utils import WGS84
+import update_vars
+from update_vars import route_analysis_date_list, CONFIG_DICT
+from segment_speed_utils.project_vars import (
+    PROJECT_CRS,
+    SEGMENT_GCS,
+    RT_SCHED_GCS,
+    analysis_date,
+)
+from segment_speed_utils import gtfs_schedule_wrangling, helpers, wrangle_shapes
+import rt_v_scheduled_trip
+
+# Times
+import datetime
+import sys
+from loguru import logger
+
+# cd rt_segment_speeds && pip install -r requirements.txt && cd ../_shared_utils && make setup_env
+# cd ../rt_scheduled_v_ran/scripts && python rt_v_scheduled_routes.py
+
+def generate_date(analysis_date:list)->str:
+    """
+    Manipulate the list of analysis dates
+    into a string that can go into a file name
+    """
+    if len(analysis_date) >= 2:
+        first_date = analysis_date[0]
+        last_date = analysis_date[-1]
+        my_string = first_date.replace('-', '_') + '_to_' + last_date.replace('-', '_')
+        return my_string
+    else:
+        return analysis_date[0]
+    
+def concatenate_trip_segment_speeds(analysis_date_list: list) -> pd.DataFrame:
+    """
+    Concatenate the trip parquets together,
+    whether it's for single day or multi-day averages.
+    """
+    TRIP_EXPORT = CONFIG_DICT["trip_metrics"]
+    df = pd.concat(
+        [
+            pd.read_parquet(
+                f"{RT_SCHED_GCS}{TRIP_EXPORT}/trip_{analysis_date}.parquet"
+            ).assign(service_date=pd.to_datetime(analysis_date))
+            for analysis_date in analysis_date_list
+        ],
+        axis=0,
+        ignore_index=True,
+    )
+    return df
+
+def route_metrics(analysis_date_list: list) -> pd.DataFrame:
+
+    df = concatenate_trip_segment_speeds(analysis_date_list)
+    # Delete out trip generated metrics
+    del_cols = [
+        "pings_per_min",
+        "spatial_accuracy_pct",
+        "rt_w_gtfs_pct",
+        "rt_v_scheduled_time_pct",
+    ]
+
+    df = df.drop(columns=del_cols)
+
+    # Add weighted metrics
+    sum_cols = [
+        "total_min_w_gtfs",
+        "rt_service_min",
+        "total_pings",
+        "service_minutes",
+        "total_vp",
+        "vp_in_shape",
+    ]
+
+    count_cols = ["trip_instance_key"]
```

Comment:
@tiffanychu90 Can you tell me if I added too many columns in the group_cols list?

Reply:
I think you just need `group_cols = ["schedule_gtfs_dataset_key", "route_id", "direction_id"]`. 
I think `peak_offpeak` and `time_of_day` are going to force the granularity to be more fine than we want. It'll leave rows like `route A - direction 0- AM_peak - peak`, `route A-direction 0 - Midday-offpeak`. which is too granular. We want `route A - direction 0 - peak`, `route A - direction 0 - offpeak`, `route A - direction 0 - all_day`